e2e/rbd.go

```diff
@@ -263,33 +263,90 @@ var _ = Describe("RBD", func() {
 						e2elog.Logf("backend image count %d expected image count %d", len(images), 1)
 						Fail("validate backend image failed")
 					}
+					totalCount := 5
 					snap := getSnapshot(snapshotPath)
 					snap.Namespace = f.UniqueName
 					snap.Spec.Source.PersistentVolumeClaimName = &pvc.Name
-					err = createSnapshot(&snap, deployTimeout)
+					// create snapshot
+					for i := 0; i < totalCount; i++ {
+						snap.Name = fmt.Sprintf("%s%d", f.UniqueName, i)
+						err = createSnapshot(&snap, deployTimeout)
+						if err != nil {
+							Fail(err.Error())
+						}
+					}
+
+					imageList := listRBDImages(f)
+					// total images in cluster is 1 parent rbd image+ total snaps
+					if len(imageList) != totalCount+1 {
+						e2elog.Logf("backend images not matching kubernetes pvc,snap count,image count %d kubernetes resource count %d", len(imageList), totalCount+1)
+						Fail("validate backend images failed")
+					}
+
+					pvcClone, err := loadPVC(pvcClonePath)
 					if err != nil {
 						Fail(err.Error())
 					}
-					pool := defaultRBDPool
-					snapList, err := listSnapshots(f, pool, images[0])
+					appClone, err := loadApp(appClonePath)
 					if err != nil {
 						Fail(err.Error())
 					}
-					if len(snapList) != 1 {
-						e2elog.Logf("backend snapshot not matching kubernetes snap count,snap count = % kubernetes snap count %d", len(snapList), 1)
-						Fail("validate backend snapshot failed")
+					// create clone and bind it to an app
+					for i := 0; i < totalCount; i++ {
+						name := fmt.Sprintf("%s%d", f.UniqueName, i)
+						pvcClone.Spec.DataSource.Name = name
+						pvcClone.Namespace = f.UniqueName
+						appClone.Namespace = f.UniqueName
+						err = createPVCAndApp(name, f, pvcClone, appClone, deployTimeout)
+						if err != nil {
+							Fail(err.Error())
+						}
 					}
 
-					validatePVCAndAppBinding(pvcClonePath, appClonePath, f)
-
-					err = deleteSnapshot(&snap, deployTimeout)
-					if err != nil {
-						Fail(err.Error())
+					imageList = listRBDImages(f)
+					// total images in cluster is 1 parent rbd image+ total
+					// snaps+ total clones
+					if len(imageList) != totalCount+totalCount+1 {
```

Comment:
For clarity , we could make use of `totalSnapCount` or `totalCloneCount` variables instead of relying on `totalCount`

Reply:
@Madhu-1 just to confirm these changes are tracked for next set of PRs ?

Reply:
@humblec changes are already done